File: js/emit_vector.js
function
emit_vector(u, p)
{
	var i, n, s, v;

	n = p.dim[0]; // number of rows

	// span

	s = 1;

	for (i = 1; i < p.dim.length; i++)
		s *= p.dim[i];

	v = {type:TABLE, n:n, m:1, a:[], level:0};

	for (i = 0; i < n; i++)
		emit_matrix(v, p, 1, i * s);

	emit_update_table(v);

	u.a.push(v);
}

File: app/emit_vector.c
#include "app.h"

void
emit_vector(struct atom *p)
{
	int i, n, span;

	// compute element span

	span = 1;

	n = p->u.tensor->ndim;

	for (i = 1; i < n; i++)
		span *= p->u.tensor->dim[i];

	n = p->u.tensor->dim[0]; // number of rows

	for (i = 0; i < n; i++)
		emit_matrix(p, 1, i * span);

	emit_update_table(n, 1); // n rows, 1 column
}
